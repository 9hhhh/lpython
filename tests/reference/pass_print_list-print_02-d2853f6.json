{
    "basename": "pass_print_list-print_02-d2853f6",
    "cmd": "lpython --pass=print_list --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/print_02.py",
    "infile_hash": "481b106ee31db37a559e38991878958a6f6e72d86cdf2cd3e5960b0e",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_print_list-print_02-d2853f6.stdout",
    "stdout_hash": "01f1192433a109ab98cf60d8463c8979adbffeac3808104317572992",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}