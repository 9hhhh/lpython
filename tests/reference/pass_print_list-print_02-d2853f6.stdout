(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Function (SymbolTable 4 {}) _lpython_main_program [] [] [(SubroutineCall 1 f () [] ())] () Source Public Implementation () .false. .false. .false. .false.), f: (Function (SymbolTable 2 {__list_iterator: (Variable 2 __list_iterator Local () () Default (Integer 4 []) Source Public Required .false.), __list_iterator1: (Variable 2 __list_iterator1 Local () () Default (Integer 4 []) Source Public Required .false.), __list_iterator2: (Variable 2 __list_iterator2 Local () () Default (Integer 4 []) Source Public Required .false.), __list_iterator3: (Variable 2 __list_iterator3 Local () () Default (Integer 4 []) Source Public Required .false.), __list_iterator4: (Variable 2 __list_iterator4 Local () () Default (Integer 4 []) Source Public Required .false.), __list_iterator5: (Variable 2 __list_iterator5 Local () () Default (Integer 4 []) Source Public Required .false.), a: (Variable 2 a Local () () Default (List (Character 1 -2 () [])) Source Public Required .false.), b: (Variable 2 b Local () () Default (List (Integer 4 [])) Source Public Required .false.), c: (Variable 2 c Local () () Default (List (Real 8 [])) Source Public Required .false.), d: (Variable 2 d Local () () Default (List (Integer 4 [])) Source Public Required .false.)}) f [] [] [(= (Var 2 a) (ListConstant [(StringConstant "ab" (Character 1 2 () [])) (StringConstant "abc" (Character 1 3 () [])) (StringConstant "abcd" (Character 1 4 () []))] (List (Character 1 2 () []))) ()) (= (Var 2 b) (ListConstant [(IntegerConstant 1 (Integer 4 [])) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 3 (Integer 4 [])) (IntegerConstant 4 (Integer 4 []))] (List (Integer 4 []))) ()) (= (Var 2 c) (ListConstant [(RealConstant 1.230000 (Real 8 [])) (RealConstant 324.300000 (Real 8 [])) (RealConstant 56.431000 (Real 8 [])) (RealConstant 90.500000 (Real 8 [])) (RealConstant 34.100000 (Real 8 []))] (List (Real 8 []))) ()) (= (Var 2 d) (ListConstant [] (List (Integer 4 []))) ()) (Print () [(StringConstant "[" (Character 1 1 () []))] () (StringConstant "" (Character 1 0 () []))) (DoLoop ((Var 2 __list_iterator) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 2 a) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 2 __list_iterator) Lt (IntegerBinOp (ListLen (Var 2 a) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Logical 4 []) ()) [(Print () [(StringConstant "'" (Character 1 1 () [])) (ListItem (Var 2 a) (Var 2 __list_iterator) (Character 1 -2 () []) ()) (StringConstant "'" (Character 1 1 () []))] (StringConstant "" (Character 1 0 () [])) (StringConstant ", " (Character 1 2 () [])))] [(Print () [(StringConstant "'" (Character 1 1 () [])) (ListItem (Var 2 a) (Var 2 __list_iterator) (Character 1 -2 () []) ()) (StringConstant "'" (Character 1 1 () []))] (StringConstant "" (Character 1 0 () [])) (StringConstant "" (Character 1 0 () [])))])]) (Print () [(StringConstant "]" (Character 1 1 () []))] () ()) (Print () [(StringConstant "[" (Character 1 1 () []))] () (StringConstant "" (Character 1 0 () []))) (DoLoop ((Var 2 __list_iterator1) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 2 b) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 2 __list_iterator1) Lt (IntegerBinOp (ListLen (Var 2 b) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Logical 4 []) ()) [(Print () [(ListItem (Var 2 b) (Var 2 __list_iterator1) (Integer 4 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant ", " (Character 1 2 () [])))] [(Print () [(ListItem (Var 2 b) (Var 2 __list_iterator1) (Integer 4 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant "" (Character 1 0 () [])))])]) (Print () [(StringConstant "]" (Character 1 1 () []))] () ()) (Print () [(StringConstant "[" (Character 1 1 () []))] () (StringConstant "" (Character 1 0 () []))) (DoLoop ((Var 2 __list_iterator2) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 2 c) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 2 __list_iterator2) Lt (IntegerBinOp (ListLen (Var 2 c) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Logical 4 []) ()) [(Print () [(ListItem (Var 2 c) (Var 2 __list_iterator2) (Real 8 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant ", " (Character 1 2 () [])))] [(Print () [(ListItem (Var 2 c) (Var 2 __list_iterator2) (Real 8 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant "" (Character 1 0 () [])))])]) (Print () [(StringConstant "]" (Character 1 1 () []))] () ()) (Print () [(StringConstant "[" (Character 1 1 () []))] () (StringConstant "" (Character 1 0 () []))) (DoLoop ((Var 2 __list_iterator3) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 2 d) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 2 __list_iterator3) Lt (IntegerBinOp (ListLen (Var 2 d) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Logical 4 []) ()) [(Print () [(ListItem (Var 2 d) (Var 2 __list_iterator3) (Integer 4 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant ", " (Character 1 2 () [])))] [(Print () [(ListItem (Var 2 d) (Var 2 __list_iterator3) (Integer 4 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant "" (Character 1 0 () [])))])]) (Print () [(StringConstant "]" (Character 1 1 () []))] () ()) (Print () [(StringConstant "[" (Character 1 1 () []))] () (StringConstant "" (Character 1 0 () []))) (DoLoop ((Var 2 __list_iterator4) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (ListConstant [(IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 []))] (List (Integer 4 []))) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 2 __list_iterator4) Lt (IntegerBinOp (ListLen (ListConstant [(IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 []))] (List (Integer 4 []))) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Logical 4 []) ()) [(Print () [(ListItem (ListConstant [(IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 []))] (List (Integer 4 []))) (Var 2 __list_iterator4) (Integer 4 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant ", " (Character 1 2 () [])))] [(Print () [(ListItem (ListConstant [(IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 []))] (List (Integer 4 []))) (Var 2 __list_iterator4) (Integer 4 []) ())] (StringConstant "" (Character 1 0 () [])) (StringConstant "" (Character 1 0 () [])))])]) (Print () [(StringConstant "]" (Character 1 1 () []))] () ()) (Print () [(StringConstant "[" (Character 1 1 () []))] () (StringConstant "" (Character 1 0 () []))) (DoLoop ((Var 2 __list_iterator5) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (ListConstant [(StringConstant "a" (Character 1 1 () [])) (StringConstant "b" (Character 1 1 () [])) (StringConstant "c" (Character 1 1 () [])) (StringConstant "d" (Character 1 1 () [])) (StringConstant "e" (Character 1 1 () [])) (StringConstant "f" (Character 1 1 () []))] (List (Character 1 1 () []))) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(If (IntegerCompare (Var 2 __list_iterator5) Lt (IntegerBinOp (ListLen (ListConstant [(StringConstant "a" (Character 1 1 () [])) (StringConstant "b" (Character 1 1 () [])) (StringConstant "c" (Character 1 1 () [])) (StringConstant "d" (Character 1 1 () [])) (StringConstant "e" (Character 1 1 () [])) (StringConstant "f" (Character 1 1 () []))] (List (Character 1 1 () []))) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Logical 4 []) ()) [(Print () [(StringConstant "'" (Character 1 1 () [])) (ListItem (ListConstant [(StringConstant "a" (Character 1 1 () [])) (StringConstant "b" (Character 1 1 () [])) (StringConstant "c" (Character 1 1 () [])) (StringConstant "d" (Character 1 1 () [])) (StringConstant "e" (Character 1 1 () [])) (StringConstant "f" (Character 1 1 () []))] (List (Character 1 1 () []))) (Var 2 __list_iterator5) (Character 1 1 () []) ()) (StringConstant "'" (Character 1 1 () []))] (StringConstant "" (Character 1 0 () [])) (StringConstant ", " (Character 1 2 () [])))] [(Print () [(StringConstant "'" (Character 1 1 () [])) (ListItem (ListConstant [(StringConstant "a" (Character 1 1 () [])) (StringConstant "b" (Character 1 1 () [])) (StringConstant "c" (Character 1 1 () [])) (StringConstant "d" (Character 1 1 () [])) (StringConstant "e" (Character 1 1 () [])) (StringConstant "f" (Character 1 1 () []))] (List (Character 1 1 () []))) (Var 2 __list_iterator5) (Character 1 1 () []) ()) (StringConstant "'" (Character 1 1 () []))] (StringConstant "" (Character 1 0 () [])) (StringConstant "" (Character 1 0 () [])))])]) (Print () [(StringConstant "]" (Character 1 1 () []))] () ())] () Source Public Implementation () .false. .false. .false. .false.), main_program: (Program (SymbolTable 3 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())])}) [])
